File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        int[] count = new int[n + 1];
        List<Integer> ans = new ArrayList<>();

        for (int num : nums) {
            count[num]++;
        }

        for (int i = 1; i < n + 1; i++) {
            if (count[i] == 0) ans.add(i);
        }

        return ans;
    }
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.scala
object Solution {
    def findDisappearedNumbers(nums: Array[Int]): List[Int] = {
        var ans = List.newBuilder[Int]
        val n = nums.length
        val count = new Array[Int](n + 1)

        for (i <- 0 until n) {
            count(nums(i)) += 1
        } 

        for (i <- 1 until n + 1) {
            if (count(i) == 0) ans += i
        }

        ans.result()
    }
}